timesketch/frontend/src/components/Sketch/TimeFilter.vue

```diff
@@ -16,113 +16,219 @@ limitations under the License.
+          </p>
+        </div>
+        <div> </div>
         <div class="field">
           <p class="control">
-            <input class="input" ref="endInput" v-model="endDateTime" type="text" placeholder="2019-07-07T10:00:01" v-on:keyup.enter="selectedChip ? update() : submit()">
+            <span class="select">
+              <select v-model="selectedInterval" :disabled="!isSelected('interval')" v-on:change="offsetStart && formatDateTime()">
+                <option v-for="option in intervals" :value="option.value" :key="option.value">{{ option.text }}</option>
+              </select>
+            </span>
           </p>
         </div>
       </div>
     </div>
 
+    <!-- Time range -->
     <div class="field is-horizontal">
-        <div class="field is-grouped">
+      <div class="field-body" style="display:flex; align-items:center;">
+
+        <b-radio v-model="radio" native-value="range"></b-radio>
+        <div class="field">
           <p class="control">
-            <a :disabled="!startDateTime" class="button is-light" v-on:click="formatDateTime()">
-              <span class="icon is-small">
-                <i class="fas fa-magic"></i>
-              </span>
-              <span>Format</span>
-            </a>
+            <input class="input" ref="startInput" v-model.trim="startDateTime" type="text" :disabled="!isSelected('range')" :placeholder="getPlaceholder('range')"
+            v-on:change="startDateTime && formatDateTime()" v-on:keyup.enter="endDateTime ? submit() : jumpTo('endInput')">
           </p>
+        </div>
+
+        <span style="margin-right:10px;">&rarr;</span>
+
+        <div class="field">
           <p class="control">
-            <button v-if="selectedChip" :disabled="!(startDateTime && endDateTime)" class="button is-success is-outlined" v-on:click="update">Update</button>
-            <button v-else :disabled="!(startDateTime && endDateTime)" class="button is-success is-outlined" v-on:click="submit">+ Add time range</button>
+            <input class="input" ref="endInput" v-model.trim="endDateTime" type="text" :disabled="!isSelected('range')" :placeholder="getPlaceholder('range')"
+            v-on:change="endDateTime && formatDateTime()" v-on:keyup.enter="submit()">
           </p>
         </div>
+      </div>
+    </div>
+
+    <!-- Submit button -->
+    <div class="field is-horizontal">
+      <div class="field is-grouped">
+        <p class="control">
+          <button :disabled="!hasAllInputs()" class="button is-success" v-on:click="submit">{{ selectedChip ? 'Update' : 'Create' }}</button>
+        </p>
+      </div>
     </div>
 
   </div>
 </template>
 
 <script>
 export default {
-  props: ['start', 'end', 'selectedChip'],
+  props: ['selectedChip'],
   data () {
     return {
-      startDateTime: this.start,
-      endDateTime: this.end,
-      chip: this.selectedChip
+      startDateTime: '',
+      endDateTime: '',
+      chip: this.selectedChip,
```

Comment:
Seeing the passed chip (selectedChip) is an object this can lead to some unpredictable behavious because any update to the object in this component will propegate up to the parent component. I noticed that you do an explicit $emit later on in the code, so I assume that you want a local copy of the prop here instead?

I suggest you use spread to copy the prop into a local one:
```
chip: {...this.selectedChip}
```

Reply:
What a great suggestion, thanks! There was indeed some code that was modifying the object that was passed in via the prop.